topostats/run_topostats.py

```diff
@@ -228,16 +228,30 @@ def main(args=None):
 
     with Pool(processes=config["cores"]) as pool:
         results = defaultdict()
+        image_stats_df_list = []
         with tqdm(
             total=len(img_files),
             desc=f"Processing images from {config['base_dir']}, results are under {config['output_dir']}",
         ) as pbar:
-            for img, result in pool.imap_unordered(
+            for img, result, image_stats_df in pool.imap_unordered(
                 processing_function,
                 scan_data_dict.values(),
             ):
                 results[str(img)] = result
                 pbar.update()
+
+                # Add the dataframe to the list for later concatenation
+                image_stats_df_list.append(image_stats_df)
```

Comment:
To align with above suggestion...
```suggestion
                image_stats[str(img)] = image_stats_df
```